src/main.py

```diff
@@ -34,27 +34,33 @@ def flush(self):
 # initialize logging before loading any third-party modules, as they may cause logging to get configured.
 logging.basicConfig(level=logging.DEBUG)
 
-# make sure we add Kolibri's dist folder to the path early on so that we avoid import errors
-if getattr(sys, 'frozen', False):
-    root_dir = os.path.dirname(os.path.abspath(sys.executable))
-else:
-    root_dir = os.path.dirname(os.path.abspath(__file__))
-locale_root_dir = os.path.join(root_dir, 'locale')
-if root_dir.endswith('src'):
-    locale_root_dir = os.path.join(root_dir, '..', 'locale')
-
+# Set the root_dir to the path where assets and locale dirs are
 if getattr(sys, 'frozen', False):
+    # In the app bundle context sys.executable will be:
+    #   On Windows: The Kolibri.exe path
+    #   On macOS: Kolibri.app/Contents/MacOS/python
     if sys.platform == 'darwin':
-        # On Mac, included Python packages go into the lib/python3.6
-        root_dir = os.path.join(root_dir, "lib", "python3.6")
-
-    # make sure we send all app output to logs as we have no console to view them on.
+        root_dir = os.path.abspath(os.path.join(os.path.dirname(sys.executable), '..', 'Resources'))
```

Comment:
Can we adjust `root_dir` on Mac to be `os.path.join(os.path.dirname(sys.executable), '..', '..', '..')`, then adjust the definition of `extra_python_path` accordingly? This way, `root_dir` is always the dir containing the .app / .exe, and we don't have to account for `root_dir` pointing to different directories on Windows and Mac. (e.g. when setting `portable_dirs`)

Reply:
Actually, even when the portable directories may vary, the `root_dir` definition is still consistent: "the path where assets and locale dirs are". That means if we opt for the `root_dir` to be always the dir containing the .app / .exe, then we have to adjust the `assets` and `locale` dirs for each platform, like moving the problem to a different place.

But I'm good for changing the definition of `root_dir`.

Reply:
Ah, good point, and another example of why my choice of `root_dir` was confusing. ;-) Maybe what we need is `src_root_dir` and `app_dir`?

Reply:
Sure!, that's better